### PR TITLE
`[Exiled::API]` Rework MirrorExtension

### PR DESCRIPTION
**Pull request by NotZer0Two**
Removed all Mirror Extension from MirrorExtensions.cs and Positioned Them inside the classes Extension

### CODE REVIEW

**Comment by NotZer0Two** [comment]
Im reworking it until next update don't review it

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -3923,6 +3924,119 @@ public void SetRank(string rankName, string rankColor)
...
+        /// <param name="target">Target to set info.</param>
+        /// <param name="info">Setting info.</param>
+        public void SetPlayerInfoForTargetOnly(Player target, string info) => MirrorExtensions.SendFakeTargetRpc(ReferenceHub, target.ReferenceHub.networkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_customPlayerInfoString), info);
+
+        /// <summary>
+        /// Sets <see cref="Player.DisplayNickname"/> of the player that only the <paramref name="target"/> player can see.
+        /// </summary>
+        /// <param name="player">Player that will desync the CustomName.</param>
+        /// <param name="name">Nickname to set.</param>
+        public void SetName(Player target, string name) => target.SendFakeSyncVar(NetworkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_displayName), name);
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Player.cs:3966]:
```suggestion
        public void SetNameForTargetOnly(Player target, string name) => target.SendFakeSyncVar(NetworkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_displayName), name);
```

File: Exiled.API/Features/Cassie.cs [hunk 2]
```diff
@@ -184,5 +184,55 @@ public static void CustomScpTermination(string scpName, CustomHandlerBase info)
...
+
+        /// <summary>
+        /// Send CASSIE announcement that only <see cref="Player"/> can hear.
+        /// </summary>
+        /// <param name="player">Target to send.</param>
+        /// <param name="words">Announcement words.</param>
+        /// <param name="makeHold">Same on <see cref="Cassie.Message(string, bool, bool, bool)"/>'s isHeld.</param>
+        /// <param name="makeNoise">Same on <see cref="Cassie.Message(string, bool, bool, bool)"/>'s isNoisy.</param>
+        /// <param name="isSubtitles">Same on <see cref="Cassie.Message(string, bool, bool, bool)"/>'s isSubtitles.</param>
+        public static void PlayCassieAnnouncement(this Player player, string words, bool makeHold = false, bool makeNoise = true, bool isSubtitles = false)
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Cassie.cs:196]: Add it  to the `Player` class.

File: Exiled.API/Features/Room.cs [hunk 3]
```diff
@@ -514,5 +514,20 @@ private static ZoneType FindZone(GameObject gameObject)
...
             return ZoneType.Unspecified;
         }
+
+        /// <summary>
+        /// Sets <see cref="Room.Color"/> of the room that only the <paramref name="target"/> player can see.
+        /// </summary>
+        /// <param name="room">Room to modify.</param>
+        /// <param name="target">Only this player can see room color.</param>
+        /// <param name="color">Color to set.</param>
+        public void SetRoomColorForTargetOnly(Player target, Color color) => target.SendFakeSyncVar(RoomLightControllerNetIdentity, typeof(RoomLightController), nameof(RoomLightController.NetworkOverrideColor), color);
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Room.cs:524]:
```suggestion
        public void SetColorForTargetOnly(Player target, Color color) => target.SendFakeSyncVar(RoomLightControllerNetIdentity, typeof(RoomLightController), nameof(RoomLightController.NetworkOverrideColor), color);
```

File: Exiled.API/Features/Room.cs [hunk 4]
```diff
@@ -514,5 +514,20 @@ private static ZoneType FindZone(GameObject gameObject)
...
+
+        /// <summary>
+        /// Sets the lights of the room to be either on or off, visible only to the <paramref name="target"/> player.
+        /// </summary>
+        /// <param name="target">The player who will see the lights state change.</param>
+        /// <param name="value">The state to set the lights to. True for on, false for off.</param>
+        public void SetRoomLightsForTargetOnly(Player target, bool value) => target.SendFakeSyncVar(RoomLightControllerNetIdentity, typeof(RoomLightController), nameof(RoomLightController.NetworkLightsEnabled), value);
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Room.cs:531]:
```suggestion
        public void SetLightsForTargetOnly(Player target, bool value) => target.SendFakeSyncVar(RoomLightControllerNetIdentity, typeof(RoomLightController), nameof(RoomLightController.NetworkLightsEnabled), value);
```

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Comment by NotZer0Two** [comment]
Strange i can build them but the bot cannot

File: Exiled.API/Features/Server.cs [hunk 5]
```diff
@@ -180,6 +180,11 @@ public static bool IsWhitelisted
             set => ServerConsole.WhiteListEnabled = value;
         }
 
+        /// <summary>
+        /// Gets the List of player currently whitelisted.
+        /// </summary>
+        public static HashSet<string> UsersWhitelisted => WhiteList.Users;
```
Comment (IkaOverride) [inline Exiled.API/Features/Server.cs:186]: `WhitelistedPlayers` or `WhitelistedUsers` plsplspls

**Pull request merged**